ext/fileinfo/libmagic/apprentice.c

```diff
@@ -632,8 +632,6 @@ file_apprentice(struct magic_set *ms, const char *fn, int action)
 		fn = p;
 	}
 
-	efree(mfn);
```

Comment:
Not sure to understand what you mean. I see that you need to save `fn` after `apprentice_1` call, it still needs to be freed at some point.